cost/google/cud_report/Google_Committed_UseDiscount_Report.pt

```diff
@@ -0,0 +1,134 @@
+  type "string"
+  label "Google Cloud Project"
+  allowed_pattern /^[0-9a-z:\.-]+$/
+end
+
+parameter "param_cud_status" do
+ label "CUD status"
+ type "string"
+ allowed_values "All","Active","Expired"
+ default "All"
+end
+
+#auth type: Oauth2, using JWT token.
+auth "auth_google", type: "oauth2" do
+  token_url "https://oauth2.googleapis.com/token"
+  grant type: "jwt_bearer" do
+  iss cred("GC_SA_CLIENT_EMAIL")
+  aud "https://oauth2.googleapis.com/token"
+  additional_claims do {
+      "scope" => "https://www.googleapis.com/auth/compute"
+    } end
+  signing_key cred("GC_SA_PRIVATE_KEY")
+  end
+end
+
+pagination "google_pagination" do
+  get_page_marker do
+    body_path "{{nextPageToken}}"
+  end
+  set_page_marker do
+    query "pageToken"
+  end
+end
+
+#https://cloud.google.com/compute/docs/instances/signing-up-committed-use-discounts#viewing_active_commitments
+datasource "ds_committed_use_discount" do
+  request do
+    auth $auth_google
+    host "www.googleapis.com"
+  header "Content-Type", "application/json"
+    path join(["/compute/v1/projects/",$param_google_project,"/aggregated/commitments"])
+    query "accept" , "application/json"
+  end
+
+  result do
+  encoding "json"
+  collect jmes_path(response, "items.*.commitments[0]") do
+  field "cud_id", jmes_path(col_item, 'id')
+  field "name", jmes_path(col_item, 'name')
+  field "region", jmes_path(col_item, 'region')
+  field "status", jmes_path(col_item, 'status')
+  field "plan", jmes_path(col_item, 'plan')
+  field "endTimestamp", jmes_path(col_item, 'endTimestamp')
+  field "startTimestamp", jmes_path(col_item, 'startTimestamp')
+  end
+ end
+end
+
+datasource "ds_committed_use_discount_map" do
+  run_script $js_committed_use_discount_map, $ds_committed_use_discount, $param_cud_status
+end
+
+#Filter the JSON, based on user selection for param 'param_cud_status' and formating Date.
+script "js_committed_use_discount_map", type: "javascript" do
+  parameters "commitments","param_cud_status"
+  result "content"
+  code <<-EOS
+  console.log(commitments)
+  var content=[];
+  for(var i=0; i < commitments.length ; i++){
+  commitment = commitments[i]
+
+  var start_date = (new Date(commitment['startTimestamp'])).toISOString().slice(0, 10);
+  var end_date = (new Date(commitment['endTimestamp'])).toISOString().slice(0, 10);
+
+  commitmentDetails={cud_id: commitment['cud_id'],
+            name: commitment['name'],
+            region: commitment['region'],
+            status: commitment['status'],
+            startTimestamp: start_date,
+            plan: commitment['plan'],
+            endTimestamp: end_date
+          }
+
+  if(param_cud_status != 'All'){
+    if(commitments[i].status === param_cud_status.toUpperCase()){
+    content.push(commitmentDetails)
+    }
+  }else{
+    content.push(commitmentDetails)
+  }
+  }
+EOS
+end
+
+
+escalation "report_list_of_CUDs" do
+   email $param_email
+end
+
+policy "policy_committed_use_discount" do
```

Comment:
Recommend we update the wording in the policy template piece to remove RS Account from the subject as its not pertinent to the incident data. By moving it to the bottom, it still allows the person receiving the email to know where the policy is applied without being the focus of the email. Also added some additional tweaks by adding the project and the status type to the subject.

Something like this:
```
policy "policy_committed_use_discount" do
  validate $ds_committed_use_discount_map do
    summary_template "{{ parameters.param_google_project }} : List of {{ parameters.param_cud_status }} Committed Use Discounts in Google Cloud"
    detail_template <<-EOS
# {{ parameters.param_cud_status }} Committed Use Discounts in Google Cloud for Project: {{ parameters.param_google_project }}

| CUD Id |  Name  | Region | Status | Plan | Start Timestamp | End Timestamp |
| ------ | ------ | ------ | ------ | ---- | --------------- | ------------- |
{{ range data -}}
| {{.cud_id}} | {{.name}} | {{.region}} | {{.status}} | {{.plan}} | {{.startTimestamp}} | {{.endTimestamp}} |
{{ end -}}

___
###### Policy Applied in Account: {{ rs_project_name }} (Account ID: {{ rs_project_id }}) within Org: {{ rs_org_name }} (Org ID: {{ rs_org_id }})
EOS
  escalate $report_list_of_CUDs
  check eq(size(data),0)
  end
end```

Reply:
@ryanoleary seeing as you just asked for this to be added, would like your feedback/ok here.

Reply:
I talked to @ryanoleary about this, and he agrees. Google project should be in the `summary_template` and the `detail_template`.

Reply:
@srpomeroy  I have made the required changes.